apis/clusters/v1beta1/postgresql_types.go

```diff
@@ -697,3 +701,25 @@ func (pg *PostgreSQL) GetHeadlessPorts() []k8scorev1.ServicePort {
 	}
 	return headlessPorts
 }
+
+// PgExtension defines desired state of an extension
+type PgExtension struct {
+	Name    string `json:"name"`
+	Enabled bool   `json:"enabled"`
+}
```

Comment:
Why don't we validate any extensions? Will we rely on the error thrown by the Instaclustr API?

Reply:
I didn't get what you meant. If we're talking about extensions immutability then they are immutable by k8s validation rules. If we're talking about describing which extensions are available, only `PgVector` is available currently. However, it may change so I decided not to handle this case.